adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -1777,8 +1777,15 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
 
     (optPredicate, optProjection) match {
       case (None, None) => {
-        ParquetUnboundAlignmentRecordRDD(sc, pathName, sd, rgd, pgs)
+        val hiveFlagFile = new File(pathName, "_hivePartitioned")
```

Comment:
This will only work for a file that is stored on a "local" (read: mounted POSIX compatible) file system. I'd suggest factoring this out into a function:

```
private def checkHiveFlag(fileName: String): Boolean = {
  val hivePath = new Path(pathName, "_hivePartitioned")
  val fs = hivePath.getFileSystem(sc.hadoopConfiguration)
  fs.exists(hivePath)
}
```